src/jdk.jfr/share/classes/jdk/jfr/internal/EventWriter.java

```diff
@@ -129,7 +129,7 @@ public void putString(String s, StringPool pool) {
         }
         if (length > StringPool.MIN_LIMIT && length < StringPool.MAX_LIMIT) {
             long l = StringPool.addString(s);
-            if (l > 0) {
+            if (l > -1) {
```

Comment:
I cannot vouch for the correctness of this patch, but I would suggest to style new check as `if (l >= 0)`. Not only this looks cleaner, the comparisons with zero are tad more efficient.

Reply:
Thanks. I've updated following your recommendation.